src/CFBG.cpp

```diff
@@ -655,6 +671,44 @@ bool CFBG::FillPlayersToCFBG(BattlegroundQueue* bgqueue, Battleground* bg, const
 
             return true;
         }
+
+        /* only for EvenTeams */
+        uint32 c = 0;
+        uint32 sumLevel = 0;
+        uint32 sumItemLevel = 0;
+        averagePlayersLevelQueue = 0;
+        averagePlayersItemLevelQueue = 0;
+
+        BattlegroundQueue::GroupsQueueType::const_iterator Ali_itr = bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].begin();
+        while (Ali_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].end() && bgqueue->m_SelectionPools[TEAM_ALLIANCE].AddGroup((*Ali_itr), aliFree))
+        {
+            auto playerGuid = *((*Ali_itr)->Players.begin());
+            Player* player = ObjectAccessor::FindPlayer(playerGuid);
+            sumLevel += player->getLevel();
+            sumItemLevel += player->GetAverageItemLevel();
+            Ali_itr++;
+            c++;
+        }
+
+        BattlegroundQueue::GroupsQueueType::const_iterator Horde_itr = bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].begin();
+        while (Horde_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].end() && bgqueue->m_SelectionPools[TEAM_HORDE].AddGroup((*Horde_itr), hordeFree))
+        {
+            auto playerGuid = *((*Horde_itr)->Players.begin());
+            Player* player = ObjectAccessor::FindPlayer(playerGuid);
+            sumLevel += player->getLevel();
+            sumItemLevel += player->GetAverageItemLevel();
+            Horde_itr++;
+            c++;
+        }
+
+        if (c > 0 && sumLevel > 0)
```

Comment:
same as above.. what is `c` ? please use descriptive names for variables/functions